test/SlowTests/Client/Subscriptions/SubscriptionsBasic.cs

```diff
@@ -770,6 +770,295 @@ public async Task ShouldThrow()
+
+                    using (var session = store.OpenSession())
+                    {
+                        for (int i = 0; i < 2; i++)
+                            session.Store(new User()
+                            {
+                                Name = "a"
+                            }, "User/");
+
+                        for (int i = 0; i < 6; i++)
+                            session.Store(new User()
+                            {
+                                Name = "a",
+                                AddressId = "b"
+                            }, "User/");
+                        session.SaveChanges();
+                    }
+                    using (var session = store.OpenSession())
+                    {
+                        for (int i = 0; i < 3; i++)
+                            session.Store(new Person()
+                            {
+                                Name = "a"
+                            }, "Person/");
+
+                        for (int i = 0; i < 2; i++)
+                            session.Store(new Person()
+                            {
+                                Name = "b"
+                            }, "Person/");
+
+                        for (int i = 0; i < 6; i++)
+                            session.Store(new Person()
+                            {
+                                Name = "a",
+                                AddressId = "b"
+                            }, "Person/");
+                        session.SaveChanges();
+                    }
+
+                    var users = 0;
+                    var people = 0;
+                    var t = allSubscription.Run(x =>
+                    {
+                        foreach (var item in x.Items)
+                        {
+                            var type = item.Result.GetType();
+                            if (type == typeof(User))
+                                users++;
+                            else if (type == typeof(Person))
+                                people++;
+                            else
+                                Assert.True(false, $"Expected get {typeof(User).FullName}, {typeof(Person).FullName} but got {item.Result.GetType().FullName}");
+                        }
+
+                        allDocs.Signal(x.NumberOfItemsInBatch);
+                    });
+
+                    Assert.True(allDocs.Wait(_reasonableWaitTime));
+                    Assert.Equal(6, users);
+                    Assert.Equal(6, people);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task SubscriptionOnMultipleCollectionsWillReturnResultsWhenAnyOfTheCollectionsChange()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var allId = await store.Subscriptions.CreateAsync(new SubscriptionCreationOptions
+                {
+                    Query = "from (Users, People, PersonWithAddresses, GeekPeople) as doc"
```

Comment:
I don't like this approach. There is no good way for us to do more interesting stuff here.
Better to do something like multi map with providing multiple queries.
There is also the issue of how to handle this on the client side? I don't like the kind we have right now